packages/amplify-graphql-transformer-core/src/graphql-api.ts

```diff
@@ -208,6 +209,21 @@ export class GraphQLApi extends GraphqlApiBase implements GraphQLAPIProvider {
         api: this,
       });
     }
+
+    // set up log retention
+    if (props.logging) {
+      const defaultRetentionDays = RetentionDays.ONE_WEEK;
+
+      const retention =
+        typeof props.logging === 'object' && 'retention' in props.logging
+          ? props.logging.retention ?? defaultRetentionDays
+          : defaultRetentionDays;
```

Comment:
nit: consider extracting this out to `isLogConfig` (or similar naming) for readability.
```ts
function isLogConfig(logging: Logging): logging is LogConfig { 
    return typeof logging === 'object' && 'retention' in logging;
}
```